parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -71,7 +79,19 @@ public boolean equals(Object obj) {
 
   @Override
   public String toString() {
-    return "Binary{" + length() + " bytes, " + Arrays.toString(getBytes()) + "}";
+    return "Binary{" + length() + " bytes, " + Arrays.toString(getBytesUnsafe()) + "}";
+  }
+
+  public Binary copy() {
+    if (isUnmodifiable) {
+      return this;
+    } else {
+      return Binary.fromReusedByteArray(getBytes());
```

Comment:
Because this allocates a new, safe copy of the bytes and forgets about it, this should instead call the unmodifiable wrapper method.